lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,8 +71,30 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+
     fail(data['errormessage']) if !data['errormessage'].empty?
-    true
+
+    notify_reboot_pending if data['rebootrequired'] == true
+
+    data
+  end
+
+  def notify_reboot_pending
+    Puppet.info "A reboot is required to progress further. Notifying Puppet."
+
+    reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
```

Comment:
Not sure I understand the comment.  We have 2 independent things happening here:

* When we build up the graph, we draw an edge (by finding the resource)
* When we need to set the `pending` state (which is what this code does), we find the resource, and set a flag


We need the edge in the graph is to ensure we take a look at pending reboot status after each and every DSC resource application.